tensorflow/python/training/momentum_test.py

```diff
@@ -107,8 +120,9 @@ def testResourceBasic(self):
     self.doTestBasic(use_resource=True)
 
   def testNesterovMomentum(self):
-    for dtype in [dtypes.float32, dtypes.float64]:
-      with self.test_session():
+    for (dtype, test_gpu) in itertools.product(
```

Comment:
If I was doing this, I would use the test config in all the test cases with a bool option like `use_half` to add `half` to the config list when needed. This keeps things consistent across the file and also makes it explicit which tests use half and which do not.

Something like:
```
def GetTestConfigs(use_half=True):
  """Get all the valid tests configs to run.
  Args:
      use_half (bool): Whether to add dtypes.half to test config
  Returns:
    all the valid test configs as tuples of dtype and use_gpu.
  """
  test_configs = itertools.product(
      [dtypes.float32, dtypes.float64], [False, True])
  if use_half:
    test_configs += [(dtypes.half, False)]
  if use_half && test.is_gpu_available(cuda_only=True):
    # dtypes.half is currently only supported on CUDA devices not SYCL devices.
    test_configs += [(dtypes.half, True)]
  return test_configs
```
but check that this actually works as expected.